conda/core/solve.py

```diff
@@ -1394,8 +1402,11 @@ def get_pinned_specs(prefix):
 
 
 def diff_for_unlink_link_precs(
-    prefix, final_precs, specs_to_add=(), force_reinstall=NULL
-):
+    prefix,
+    final_precs,
+    specs_to_add=(),
+    force_reinstall=NULL,
+) -> tuple[tuple[PackageRecord, ...], tuple[PackageRecord, ...]]:
```

Comment:
Can we define this type at the top of the file somehow as `DiffResult` or something?

Reply:
`DiffResult` seemed to have limited usability so I opted for `tuple[PackageRecordSequence, PackageRecordSequence]` instead, see https://github.com/conda/conda/pull/13053/commits/85462f9e2b67b8028cc08c3955857736c33fae93